core/src/components/radio-group/radio-group.tsx

```diff
@@ -139,13 +150,15 @@ export class RadioGroup implements ComponentInterface {
 
         if (!inSelectPopover) {
           this.value = next.value;
+          this.emitValueChange(ev);
         }
       }
 
       // Update the radio group value when a user presses the
       // space bar on top of a selected radio
       if (['Space'].includes(ev.code)) {
         this.value = this.allowEmptySelection && this.value !== undefined ? undefined : current.value;
+        this.emitValueChange(ev);
```

Comment:
This causes `ionChange` to fire if you press Space when highlighting the radio that's already selected (when `allowEmptySelection="false"`), thus not changing the value.